src/Microsoft.Kiota.Cli.Commons/IO/ODataPagingService.cs

```diff
@@ -84,9 +97,11 @@ private static bool IsJson(PageLinkData pageLinkData)
             return left ?? right;
         }
 
-        JsonNode? nodeLeft = JsonNode.Parse(left);
+        var taskLeft = JsonNode.ParseAsync(left, cancellationToken: cancellationToken);
+        var taskRight = JsonNode.ParseAsync(right, cancellationToken: cancellationToken);
+        var nodeLeft = await taskLeft;
+        var nodeRight = await taskRight;

```

Comment:
```suggestion
        var nodeLeft = await JsonNode.ParseAsync(left, cancellationToken: cancellationToken);
        var nodeRight = await JsonNode.ParseAsync(right, cancellationToken: cancellationToken);
```

Reply:
This makes the async tasks serial, i.e. taskRight won't start until taskLeft is complete. I thought I could use concurrency since the tasks are disjoint. Awaiting after the tasks have started should allow better concurrency. I believe `ParseAsync` is a hot task, i.e starts when we call it as opposed to when we await on it.